### PR TITLE
Add service config forwarded attributes

### PR DESCRIPTION
**Pull request by kyessenov**
Implements https://github.com/istio/api/pull/492
Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: src/istio/control/http/request_handler_impl.cc [hunk 1]
```diff
@@ -48,14 +48,8 @@ CancelFunc RequestHandlerImpl::Check(CheckData* check_data,
...
   ExtractRequestAttributes(check_data);
-
-  if (service_context_->client_context()->config().has_forward_attributes()) {
-    AttributesBuilder::ForwardAttributes(
-        service_context_->client_context()->config().forward_attributes(),
-        header_update);
-  } else {
-    header_update->RemoveIstioAttributes();
-  }
+  header_update->RemoveIstioAttributes();
```
Comment (kyessenov) [inline src/istio/control/http/request_handler_impl.cc]: I think we should unconditionally remove it after extracting it, and not depend on presence of static forwarded attributes config.

File: src/istio/control/http/service_context.cc [hunk 2]
```diff
@@ -36,30 +38,50 @@ void ServiceContext::BuildParsers() {
...
     request->attributes.MergeFrom(service_config_->mixer_attributes());
   }
 }
 
-void ServiceContext::AddApiAttributes(CheckData* check_data,
-                                      RequestContext* request) const {
+// Inject a header that contains the static forwarded attributes.
+void ServiceContext::InjectForwardedAttributes(
+    HeaderUpdate *header_update) const {
+  bool forward = false;
```
Comment (qiwzhang) [inline src/istio/control/http/service_context.cc]: not need this bool,  I think you can check if result attributes is empty or not

File: src/istio/control/http/request_handler_impl.cc (hunk 1 above)
Comment (qiwzhang) [reply src/istio/control/http/request_handler_impl.cc]: +1

File: repositories.bzl [hunk 3]
```diff
@@ -188,7 +188,7 @@ filegroup(
     native.new_git_repository(
         name = "mixerapi_git",
         build_file_content = BUILD,
-        commit = ISTIO_API,
+        commit = "b33fb6d1d44ae6e7252004c8916a702976cfb101",
```
Comment (qiwzhang) [inline repositories.bzl]: you should assign new sha  to ISTIO_API, not repleace it

**Comment by qiwzhang** [comment]
Please add some tests.  Please look at https://github.com/istio/proxy/blob/master/src/istio/control/http/request_handler_impl_test.cc#L411

File: repositories.bzl (hunk 3 above)
Comment (kyessenov) [reply repositories.bzl]: Temporary, until api is unfrozen and I can merge the corresponding PR.

**Comment by kyessenov** [comment]
@qiwzhang added tests. PTAL.

**Comment by lizan** [comment]
/hold
LGTM, holding until api is unfrozen

**Pull request closed without merging**